### PR TITLE
Convert AddressPage.js to TSX.

### PR DESCRIPTION
**Pull request by toolness**
This converts `AddressPage.js` to TSX.  ~~Crucially, though, most of the address-related state fields are typed to `any`, which makes some of this incomplete.  Those fields are going to take a _lot_ of work to properly type because of how they're being used, and how they interact with some components further down in the component hierarchy (some of which are still untyped). It might ultimately be easier to just refactor the data model to be more comprehensible, rather than attempting to add types as-is.~~

Note that this introduces an explicit `throw` for the situation where an address search has no geosearch information; this essentially means that the error signature for #340 will now be different, as it will be an explicit exception rather than `Cannot get property 'bbl' of undefined or null`.

### CODE REVIEW

File: client/src/components/PropertiesList.tsx [hunk 1]
```diff
@@ -10,35 +10,17 @@ import { t } from "@lingui/macro";
...
-  rsunits2007: number;
-  rsunits2017: number;
-  openviolations: number;
-  totalviolations: number;
-  evictions: string | null;
-  ownernames: { title: string; value: string }[];
-  lastsaledate: string;
-  lastsaleamount: string;
-  lastsaleacrisid: string;
-};
```
Comment (toolness) [inline client/src/components/PropertiesList.tsx:31]: Well, at least we're getting rid of this duplicate typing.

File: client/src/containers/AddressPage.tsx [hunk 2]
```diff
@@ -1,35 +1,64 @@
...
+
+export default class AddressPage extends Component<AddressPageProps, State> {
+  constructor(props: AddressPageProps) {
     super(props);
 
     this.state = {
       searchAddress: { ...props.match.params }, // maybe this should be
       userAddr: {}, // merged together?
       hasSearched: false,
-      geosearch: {},
```
Comment (toolness) [inline client/src/containers/AddressPage.tsx:32]: I'm a bit queasy about this change but hopefully it should work.  We should probably kick the tires a bit just to make sure nothing explodes...

**Comment by sraby** [comment]
@toolness this is ready for your eyes again!

**Comment by toolness** [comment]
awesome thanks @sraby !

**Pull request merged**